### PR TITLE
AMCL to use sensor data qos for scan topic

### PR DESCRIPTION
**Pull request by mkhansen-intel**
---

## Description of contribution in a few bullet points

* Changes AMCL to use the QOS profile for sensor data when subscribing to the 'scan' topic.

This was suggested by @miguelcompany in this issue: https://github.com/ros2/rmw_fastrtps/issues/280

I tested this change and see our tests pass now with the latest FastRTPS.

### CODE REVIEW

**Comment by mkhansen-intel** [comment]
Travis CI failed but it passes the localization test on the first try.

**Comment by mkhansen-intel** [comment]
With the newest ros2 nightly build docker image, this passed CI in Travis.

@ruffsl or @crdelsey - do you have permissions to re-start the circle CI builds, I don't

**Comment by mkhansen-intel** [comment]
Trying to restart Circle CI

File: nav2_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ AmclNode::AmclNode()
     };
   set_map_srv_ = create_service<nav_msgs::srv::SetMap>("set_map", handle_set_map_callback);
 
-
+  custom_qos_profile = rmw_qos_profile_sensor_data;
```
Comment (orduno) [inline nav2_amcl/src/amcl_node.cpp]: We should probably be using the new style for specifying QoS settings. 

Here's an example

https://github.com/ros-planning/navigation2/blob/6d837b17cdd81eaddca42ffde5946fbcad9f2686/nav2_robot/src/robot.cpp#L109

However, I see you are using it for message filters and am not sure it's supported there.

**Comment by ruffsl** [comment]
> @ruffsl or @crdelsey - do you have permissions to re-start the circle CI builds, I don't

I don't have write access to this PR, so it appears I can't retrigger this with my account.
(This might fix that: [Allowing changes to a pull request branch created from a fork](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork))

image

However, be sure you've logged-in/connected your github auth, not the anonymous account circleci.com gives you for non-logged in users.  
https://circleci.com/account

File: nav2_amcl/src/amcl_node.cpp (hunk 1 above)
Comment (crdelsey) [reply nav2_amcl/src/amcl_node.cpp]: I tried this and the message filter interface doesn't support the `rclcpp::SensorDataQoS()` method like you feared.

**Pull request merged**